Access Review History Defintions (#12107)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accessreviewhistorydefinition-create.md

Create a new [accessReviewHistoryDefinition](../resources/accessreviewhistorydefinition.md) object.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|AccessReview.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|AccessReview.ReadWrite.All|

The signed-in user must also be in a directory role that permits them to read an access review to retrieve any data.  For more details, see the role and permission requirements for [access reviews](../resources/accessreviewsv2-root.md).

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /identityGovernance/accessReviews/historyDefinitions
```

## Request headers

|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

In the request body, supply a JSON representation of the [accessReviewHistoryDefinition](../resources/accessreviewhistorydefinition.md) object.

> The following table shows the properties accepted to create an [accessReviewHistoryDefinition](../resources/accessreviewhistorydefinition.md).

Reviewer: Are the properties required? If so, please change 'accepted' to 'required'.

Author: I changed it to required and optional because two of these are optional parameters.

---
title: "Create accessReviewHistoryDefinition"
description: "Create a new accessReviewHistoryDefinition object."

> author: "leherpel"

Reviewer 2: Global: Please ensure the author is in the [Microsoft Graph contacts list](https://microsoft.sharepoint.com/sites/Graph/Lists/Microsoft%20Graph%20Contacts/AllItems.aspx?CT=1606755556215&OR=OWA%2DNT&CID=798496d0%2D7bc3%2Df8db%2D49ee%2D45ab03388a04). This is the person responsible for all workload PRs and GitHub issues.

Author: Updating to isabelleatmsft

|Scenario| Query |
|--|--|
| Include every `accessReviewScheduleDefinition` review result on individual groups (excludes definitions scoped to all Microsoft 365 groups with guest users) | /identityGovernance/accessReviews/definitions?$filter=contains(scope/query, '/groups')" |
| Include every `accessReviewScheduleDefinition` review result on a specific group (excludes definitions scoped to all Microsoft 365 groups with guest users) | /identityGovernance/accessReviews/definitions?$filter=contains(scope/microsoft.graph.accessReviewQueryScope/query, '/groups/{group id}') |
| Include every `accessReviewScheduleDefinition` review result scoped to all Microsoft 365 groups with guest users | /identityGovernance/accessReviews/definitions?$filter=contains(scope/microsoft.graph.accessReviewQueryScope/query, './members') |
| Include every `accessReviewScheduleDefinition` review result on an access package | /identityGovernance/accessReviews/definitions?$filter=contains(scope/microsoft.graph.accessReviewQueryScope/query, 'accessPackageAssignments') |

> | Include every `accessReviewScheduleDefinition` review result for service principals assigned to privileged role | /identityGovernance/accessReviews/definitions?$filter=contains(scope/microsoft.graph.accessReviewQueryScope/query, 'roleAssignmentScheduleInstances') |

Reviewer 2: There's a WIP PR to transform these into a conceptual article for accessReviewQueryScope. When the changes go live, we'll revisit this and link to the page.

## Response

If successful, this method returns a `201 Created` response code and an [accessReviewHistoryDefinition](../resources/accessreviewhistorydefinition.md) object in the response body.

## Examples

> This is an example of creating an access review history definition which is scoped to access package and group reviews starting and ending on or between 01/01/2021 and 04/05/2021.

Reviewer 2: ```suggestion
This is an example of creating an access review history definition scoped to access reviews on access packages and groups, running between the start date of 01/01/2021 and end date of 04/05/2021.
```
